ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -462,6 +475,44 @@ private void expandNestedListType(String jsonFieldName, Object val, Field field,
     });
   }
 
+  /**
+   * This method gives us a chance to perform some additional type coercions in case the BigQuery
+   * field type is different from the source data type. This should rarely happen, since only
+   * validated payloads get through to this BQ sink path, but there are sets of probes with
+   * heterogeneous types that appear as explicit fields in BQ, but are treated as loosely typed
+   * maps at the validation phase; we need to catch these or they can cause the entire pipeline
+   * to stall.
+   *
+   * <p>Returning {@code null} here indicates that no coercion is defined and that the field should
+   * be put to {@code additional_properties}.
+   */
+  private static Object coerceToBqType(Object o, Field field) {
+    if (field.getMode() == Mode.REPEATED) {
+      return o;
+    } else {
+      if (field.getType() == LegacySQLTypeName.STRING) {
+        return coerceToString(o);
+      } else if (field.getType() == LegacySQLTypeName.INTEGER) {
```

Comment:
I think it would be more consistent with the existing code to add cases for `field.getType() == LegacySQLTypeName.INTEGER && field.getMode() != Mode.REPEATED` and `field.getType() == LegacySQLTypeName.BOOLEAN && field.getMode() != Mode.REPEATED` to `processField`, rather than breaking it out into a separate function. this would also make it more obvious that the case of `field.getType() == LegacySQLTypeName.STRING` is already handled.

Reply:
I'm inclined to remove both of the existing cases that do string coercion and pull that all into a single coerce method as entrypoint.